org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/AspectjFilter.java

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2018 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Lars Grefer - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.internal.analysis.filter;
+
+import org.objectweb.asm.Attribute;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.*;
+
+/**
+ * Filters synthetic methods created by the AspectJ-Compiler (ajc)
+ */
+public class AspectjFilter implements IFilter {
+
+    private static final String AJ_SYNTHETIC_ATTRIBUTE = "org.aspectj.weaver.AjSynthetic";
+
+    public void filter(MethodNode methodNode, IFilterContext context, IFilterOutput output) {
+
+        if (isAjSynthetic(methodNode)) {
+            output.ignore(methodNode.instructions.getFirst(), methodNode.instructions.getLast());
+            return;
+        }
+
+        if (methodNode.name.equals("<clinit>")) {
+            checkStaticInitializer(methodNode, output);
+        }
+
+    }
+
+    private void checkStaticInitializer(MethodNode methodNode, IFilterOutput output) {
+        AbstractInsnNode preClinitNode = null;
+
+        for (AbstractInsnNode node = methodNode.instructions.getFirst();
+             node != null;
+             node = node.getNext()) {
+
+            if (node.getOpcode() == Opcodes.INVOKESTATIC) {
```

Comment:
Inversion of condition and usage of `continue` will reduce nesting of this code and hence might improve its readability. Also note that we have `AbstractMatcher` which simplifies matching of bytecode. Maybe it can be used here?